packages/shared/src/components/fields/MarkdownInput/index.tsx

```diff
@@ -260,6 +398,8 @@ function MarkdownInput(
                 )}
                 value={input}
                 onClick={onInputClick}
+                onKeyDown={handleMarkdownShortcut}
+                onPaste={handleLinkPaste}
```

Comment:
Can you check on how we can move the logic here to the `useMarkdownInput` hook? The thing is, there are existing callbacks we pass down there at line 392. Defining these new ones would mean we overwrite the existing behaviors which involves mentioning of users and navigating the list through keyboard.